gflags now seems replacing "gflags" namespace to "google" in gflags/gflags.h

When compiling on MAC OSX 10.9.4, I encountered an error as follows :

/usr/bin/clang++ tools/caffe.cpp -stdlib=libstdc++ -pthread -fPIC -DNDEBUG -O2 -I/usr/include/python2.7 -I/usr/lib/python2.7/dist-packages/numpy/core/include -I/usr/local/include -I.build_release/src -I./src -I./include -I/usr/local/cuda/include -I/System/Library/Frameworks/vecLib.framework/Versions/Current/Headers/ -Wall -Wno-sign-compare -Wno-unneeded-internal-declaration -c -o .build_release/tools/caffe.o 2> .build_release/tools/caffe.o.warnings.txt \
        || (cat .build_release/tools/caffe.o.warnings.txt; exit 1)
tools/caffe.cpp:219:3: error: use of undeclared identifier 'gflags'
  gflags::SetUsageMessage("command line brew\n"
  ^
tools/caffe.cpp:231:5: error: use of undeclared identifier 'gflags'
    gflags::ShowUsageWithFlagsRestrict(argv[0], "tools/caffe");
    ^
2 errors generated.
make: **\* [.build_release/tools/caffe.o] Error 1

After little investigation, it seems gflags/gflags.h is using google namespace instead of gflags namespace

--- a/include/caffe/common.hpp
+++ b/include/caffe/common.hpp
@@ -2,6 +2,7 @@
 #define CAFFE_COMMON_HPP_
 
 #include <boost/shared_ptr.hpp>
+#include <gflags/gflags.h>
 #include <glog/logging.h>
 
 #include <cmath>
@@ -10,6 +11,15 @@
 #include <string>
 
 #include "caffe/util/device_alternate.hpp"
+
+// gflags 2.1 issue: namespace google was changed to gflags without warning.
+// Luckily we will be able to use GFLAGS_GFAGS_H_ to detect if it is version
+// 2.1. If yes , we will add a temporary solution to redirect the namespace.
+// TODO(Yangqing): Once gflags solves the problem in a more elegant way, let's
+// remove the following hack.
+#ifndef GFLAGS_GFLAGS_H_
+namespace gflags = google;
+#endif  // GFLAGS_GFLAGS_H_
 
 // Disable the copy and assignment operator for a class.
 #define DISABLE_COPY_AND_ASSIGN(classname) \

--- a/src/caffe/common.cpp
+++ b/src/caffe/common.cpp
@@ -1,21 +1,9 @@
-#include <gflags/gflags.h>
 #include <glog/logging.h>
 #include <cstdio>
 #include <ctime>
 
 #include "caffe/common.hpp"
 #include "caffe/util/rng.hpp"
-
-// gflags 2.1 issue: namespace google was changed to gflags without warning.
-// Luckily we will be able to use GFLAGS_GFAGS_H_ to detect if it is version
-// 2.1. If yes , we will add a temporary solution to redirect the namespace.
-// TODO(Yangqing): Once gflags solves the problem in a more elegant way, let's
-// remove the following hack.
-#ifdef GFLAGS_GFLAGS_H_
-namespace google {
-using ::gflags::ParseCommandLineFlags;
-}  // namespace google
-#endif  // GFLAGS_GFLAGS_H_
 
 namespace caffe {
 
@@ -44,7 +32,7 @@ int64_t cluster_seedgen(void) {
 
 void GlobalInit(int* pargc, char*** pargv) {
   // Google flags.
-  ::google::ParseCommandLineFlags(pargc, pargv, true);
+  ::gflags::ParseCommandLineFlags(pargc, pargv, true);
   // Google logging.
   ::google::InitGoogleLogging(*(pargv)[0]);
 }

--- a/tools/caffe.cpp
+++ b/tools/caffe.cpp
@@ -1,4 +1,3 @@
-#include <gflags/gflags.h>
 #include <glog/logging.h>
 
 #include <cstring>

